yarn-project/foundation/src/abi/abi.ts

```diff
@@ -249,6 +249,12 @@ export interface ContractArtifact {
    * The name of the contract.
    */
   name: string;
+
+  /**
+   * The version of compiler used to create this artifact
+   */
+  aztecNrVersion?: string;
```

Comment:
does this need to be optional? Shouldn't every contract have this?

Reply:
I made optional because:
- we still have to guard against contracts compiled with an old compiler that didn't do this
- our own contracts inside `yarn-project/noir-contracts` currently don't get this field set because they use relative paths to import Aztec.nr rather than git tags. Once this lands we should be able to fix this https://github.com/noir-lang/noir/pull/3427